core/src/main/java/nl/nn/adapterframework/util/MessageUtils.java

```diff
@@ -87,16 +72,52 @@ public static String computeContentType(Message message) {
 			return null;
 		}
 
-		String mimeType = (String)message.getContext().get(MessageContext.METADATA_MIMETYPE);
-		if(StringUtils.isEmpty(mimeType)) {
+		MimeType mimeType = (MimeType)message.getContext().get(MessageContext.METADATA_MIMETYPE);
+		if(mimeType == null) {
 			return null;
 		}
 
-		StringBuilder contentType = new StringBuilder(mimeType);
+		StringBuilder contentType = new StringBuilder();
+		contentType.append(mimeType.getType());
+		contentType.append('/');
+		contentType.append(mimeType.getSubtype());
+
 		if(!message.isBinary()) {
 			contentType.append(";charset=");
 			contentType.append(message.getCharset());
 		}
 		return contentType.toString();
 	}
+
+	public static MimeType getMimeType(Message message) {
+		Map<String, Object> context = message.getContext();
+		return (MimeType) context.get(MessageContext.METADATA_MIMETYPE);
+	}
+
+	public static MimeType computeMimeType(Message message, String filename) {
```

Comment:
Wanneer ga je dit gebruiken? Het zou zomaar een dure operatie kunnen zijn. Dat zou ik er wel als commentaar bij zetten.

Reply:
Het is alleen duur wanneer de message gepreserved moet worden. Alleen de file magic wordt uitgelezen < 1kb.
Als message altijd een steam zou zijn zou je het kunnen bufferen... dat je mark/reset doet na het uitlezen van de file header.